pyspider/libs/base_queue.py

```diff
@@ -0,0 +1,101 @@
+import multiprocessing
+from multiprocessing.queues import Queue as MPQueue
+from six.moves import queue as BaseQueue
+
+
+Empty = BaseQueue.Empty
+Full = BaseQueue.Full
+
+
+# The SharedCounter and Queue classes come from:
+# https://github.com/vterron/lemon/commit/9ca6b4b
+
+class SharedCounter(object):
+    """ A synchronized shared counter.
+    The locking done by multiprocessing.Value ensures that only a single
+    process or thread may read or write the in-memory ctypes object. However,
+    in order to do n += 1, Python performs a read followed by a write, so a
+    second process may read the old value before the new one is written by the
+    first process. The solution is to use a multiprocessing.Lock to guarantee
+    the atomicity of the modifications to Value.
+    This class comes almost entirely from Eli Bendersky's blog:
+    http://eli.thegreenplace.net/2012/01/04/shared-counter-with-pythons-multiprocessing/
+    """
+
+    def __init__(self, n=0):
+        self.count = multiprocessing.Value('i', n)
+
+    def increment(self, n=1):
+        """ Increment the counter by n (default = 1) """
+        with self.count.get_lock():
+            self.count.value += n
+
+    @property
+    def value(self):
+        """ Return the value of the counter """
+        return self.count.value
+
+
+class Queue(BaseQueue.Queue, object):
```

Comment:
Weird. I just tried without overriding Queue.Queue, dashboard works well. I'm pretty sure I fixed something by override it. Can't re-call my memory, and git commit history squashed already... So drop it and use `from six.moves import queue as Queue` until something wrong.